OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/KeyListFragment.java

```diff
@@ -292,25 +318,73 @@ private void startSearchForQuery() {
         // This is called when a new Loader needs to be created. This
         // sample only has one Loader, so we don't care about the ID.
         Uri uri;
-        if (!TextUtils.isEmpty(mQuery)) {
-            uri = KeyRings.buildUnifiedKeyRingsFindByUserIdUri(mQuery);
+        if (!TextUtils.isEmpty(getQuery())) {
+            uri = KeyRings.buildUnifiedKeyRingsFindByUserIdUri(getQuery());
         } else {
             uri = KeyRings.buildUnifiedKeyRingsUri();
         }
 
         // Now create and return a CursorLoader that will take care of
         // creating a Cursor for the data being displayed.
         return new CursorLoader(getActivity(), uri,
-                KeySectionedListAdapter.KeyListCursor.PROJECTION, null, null,
-                KeySectionedListAdapter.KeyListCursor.ORDER);
+                PROJECTION, null, null,
+                ORDER);
     }
 
+    public static final String ORDER = KeychainContract.KeyRings.HAS_ANY_SECRET
+            + " DESC, " + KeychainContract.KeyRings.USER_ID + " COLLATE NOCASE ASC";
+
+    public static final String[] PROJECTION = {
+            KeyRings._ID,
+            KeychainContract.KeyRings.MASTER_KEY_ID,
+            KeychainContract.KeyRings.USER_ID,
+            KeychainContract.KeyRings.IS_REVOKED,
+            KeychainContract.KeyRings.IS_EXPIRED,
+            KeychainContract.KeyRings.IS_SECURE,
+            KeychainContract.KeyRings.HAS_DUPLICATE_USER_ID,
+            KeychainContract.KeyRings.CREATION,
+            KeychainContract.KeyRings.NAME,
+            KeychainContract.KeyRings.EMAIL,
+            KeychainContract.KeyRings.COMMENT,
+            KeychainContract.KeyRings.VERIFIED,
+            KeychainContract.KeyRings.HAS_ANY_SECRET,
+            KeychainContract.KeyRings.FINGERPRINT,
+            KeychainContract.KeyRings.HAS_ENCRYPT};
+
+
     @Override
     public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
-        // Swap the new cursor in. (The framework will take care of closing the
-        // old cursor once we return.)
-        getAdapter().setSearchQuery(mQuery);
-        getAdapter().swapCursor(KeySectionedListAdapter.KeyListCursor.wrap(data));
+
+        mKeyItems.clear();
+
+        if (data.moveToFirst()) {
+            while (!data.isAfterLast()) {
+                mKeyItems.add(new KeyItem(null, KeyListCursor.wrap(data), mKeyListener));
+                data.moveToNext();
+            }
+        }
```

Comment:
@dschuermann @Valodim In my opinion it doesn't matter to fetch all keys into memory, for they won't take up much memory. A compromising way is fetching items only when they are about to showing up, by using [Endless Scroll](https://github.com/davideas/FlexibleAdapter/wiki/5.x-%7C-On-Load-More) feature of FlexibleAdapter.
On the other hand, I was told that the author of superslim has come back on development of superslim. See [this](https://github.com/TonicArtos/SuperSLiM/issues/188) for detail. ( I'm a sensitive boy :) ) So wait for the version5 of superslim is also a choice. If so, please assign the task that migrate to version5 to me :)